lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/WGMMA.cpp

```diff
@@ -170,7 +170,11 @@ DotOpMmaV3SmemLoader loadA(TritonGPUToLLVMTypeConverter *typeConverter,
 
   // The descriptor should be calculated based on the first warp of the
   // warpgroup.
-  Value warp = udiv(thread, i32_val(32));
+  Value warp = and_(udiv(thread, i32_val(32)), i32_val(0xFFFFFFFC));
+  // Workaround for a bug in ptxas 12.3 that cause a failure in
+  // test_core.py::test_dot. The shuffle will force the compiler to treate the
+  // value as uniform and prevent wrong optimizations.
```

Comment:
OOC what do you mean by treat the value as uniform ?

Reply:
meaning the value is the same for all the lanes in the warp.